frontend/src/app/modules/settings/team/edit-team/edit-team.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, EventEmitter, OnDestroy, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { BaseComponent } from '@core/base/base.component';
+import { ITeam } from '@core/models/ITeam';
+import { ConfirmationWindowService } from '@core/services/confirmation-window.service';
+import { NotificationService } from '@core/services/notification.service';
+import { SpinnerService } from '@core/services/spinner.service';
+import { TeamService } from '@core/services/team.service';
+import { TeamPreferencesComponent } from '@modules/settings/team/team-preferences/team-preferences.component';
+import { deletionMessage } from '@shared/constants/shared-messages';
+import { Subscription } from 'rxjs';
+
+@Component({
+    selector: 'app-edit-team',
+    templateUrl: './edit-team.component.html',
+    styleUrls: ['./edit-team.component.sass'],
+})
+export class EditTeamComponent extends BaseComponent implements OnDestroy {
+    private id: number | undefined;
+
+    public team: ITeam;
+
+    private deleteEventEmitter = new EventEmitter<void>();
+
+    private deleteEventSubscription: Subscription;
+
+    @ViewChild(TeamPreferencesComponent) teamPreferencesComponent: TeamPreferencesComponent;
+
+    constructor(
+        private activateRoute: ActivatedRoute,
+        private teamService: TeamService,
+        public notificationService: NotificationService,
+        private router: Router,
+        private spinnerService: SpinnerService,
+        private confirmationWindowService: ConfirmationWindowService,
+    ) {
+        super();
+        this.activateRoute.params.subscribe(params => {
+            this.id = params['id'];
+            this.spinnerService.show();
+            this.teamService.getTeamById(this.id)
+                .pipe(this.untilThis)
+                .subscribe((team) => {
+                    this.team = team;
+                    this.teamPreferencesComponent.formGroup.patchValue({
+                        name: team.name,
+                        image: team.image,
+                        pageLink: team.pageLink,
+                        timeZone: team.timeZone,
+                        description: team.description,
+                    });
+                    this.teamPreferencesComponent.imageUrl = team.image;
+                    this.spinnerService.hide();
+                });
```

Comment:
move to a separate method and handle error

Reply:
Fixed